codec-mqtt/src/main/java/io/netty/handler/codec/mqtt/MqttReasonCodes.java

```diff
@@ -0,0 +1,419 @@
+ * Provides a set of enumeration that exposes standard MQTT 5 reason codes used by various messages.
+ * */
+public final class MqttReasonCodes {
+
+    /**
+     * @return the corresponding enum value to the hex value.
+     * */
+    private static <E> E valueOfHelper(byte b, E[] values) {
+        try {
+            return values[b & 0xFF];
+        } catch (ArrayIndexOutOfBoundsException e) {
+            throw new IllegalArgumentException("unknown reason code: " + b);
+        }
+    }
+
+    /**
+     * Reason codes for MQTT Disconnect message.
+     */
+    enum Disconnect {
+        NORMAL_DISCONNECT((byte) 0x00), //sent by: client or server
+        DISCONNECT_WITH_WILL_MESSAGE((byte) 0x04), //sent by: client
+        UNSPECIFIED_ERROR((byte) 0x80), //sent by: client or server
+        MALFORMED_PACKET((byte) 0x81), //sent by: client or server
+        PROTOCOL_ERROR((byte) 0x82), //sent by: client or server
+        IMPLEMENTATION_SPECIFIC_ERROR((byte) 0x83), //sent by: client or server
+        NOT_AUTHORIZED((byte) 0x87), //sent by: server
+        SERVER_BUSY((byte) 0x89), //sent by: server
+        SERVER_SHUTTING_DOWN((byte) 0x8B), //sent by: server
+        KEEP_ALIVE_TIMEOUT((byte) 0x8D), //sent by: Server
+        SESSION_TAKEN_OVER((byte) 0x8E), //sent by: Server
+        TOPIC_FILTER_INVALID((byte) 0x8F), //sent by: Server
+        TOPIC_NAME_INVALID((byte) 0x90), //sent by: Client or Server
+        RECEIVE_MAXIMUM_EXCEEDED((byte) 0x93), //sent by: Client or Server
+        TOPIC_ALIAS_INVALID((byte) 0x94), //sent by: Client or Server
+        PACKET_TOO_LARGE((byte) 0x95), //sent by: Client or Server
+        MESSAGE_RATE_TOO_HIGH((byte) 0x96), //sent by: Client or Server
+        QUOTA_EXCEEDED((byte) 0x97), //sent by: Client or Server
+        ADMINISTRATIVE_ACTION((byte) 0x98), //sent by: Client or Server
+        PAYLOAD_FORMAT_INVALID((byte) 0x99), //sent by: Client or Server
+        RETAIN_NOT_SUPPORTED((byte) 0x9A), //sent by: Server
+        QOS_NOT_SUPPORTED((byte) 0x9B), //sent by: Server
+        USE_ANOTHER_SERVER((byte) 0x9C), //sent by: Server
+        SERVER_MOVED((byte) 0x9D), //sent by: Server
+        SHARED_SUBSCRIPTIONS_NOT_SUPPORTED((byte) 0x9E), //sent by: Server
+        CONNECTION_RATE_EXCEEDED((byte) 0x9F), //sent by: Server
+        MAXIMUM_CONNECT_TIME((byte) 0xA0), //sent by: Server
+        SUBSCRIPTION_IDENTIFIERS_NOT_SUPPORTED((byte) 0xA1), //sent by: Server
+        WILDCARD_SUBSCRIPTIONS_NOT_SUPPORTED((byte) 0xA2); //sent by: Server
+
+        protected static final Disconnect[] VALUES;
+
+        static {
+            Disconnect[] values = values();
+            VALUES = new Disconnect[28];
+            for (Disconnect code : values) {
+                final int unsignedByte = code.byteValue & 0xFF;
+                // Suppress a warning about out of bounds access since the enum contains only correct values
+                VALUES[unsignedByte] = code;    //  [java/index-out-of-bounds]
```

Comment:
This does not look correct. You initialize `VALUES` to have 28 elements, but the indexes are meant to be reason codes, which can extend to higher values because they are a sparse set. The highest index that we will assign here is 162:

```
jshell> ((byte) 0xA2) & 0xFF
$1 ==> 162
```

So the LGTM warning is correct. The code needs to be fixed, instead of the warning silenced.